src/Tools/ExternalAccess/Razor/IRazorIDocumentExcerptService.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    internal interface IRazorDocumentExcerptService
+    {
+        Task<RazorExcerptResult?> TryExcerptAsync(Document document, TextSpan span, RazorExcerptMode mode, CancellationToken cancellationToken);
+    }
+
+    internal enum RazorExcerptMode
+    {
+        SingleLine,
+        Tooltip
+    }
+
+    internal struct RazorExcerptResult
```

Comment:
>struct [](start = 13, length = 6)

`readonly`

Reply:
I'll apply the wrapper pattern to other pieces but this specific type is what Razor needs to new-up and return.